src/Microsoft.AspNetCore.OData/Batch/ODataBatchRequestItem.cs

```diff
@@ -35,19 +36,12 @@ public static async Task SendRequestAsync(RequestDelegate handler, HttpContext c
 
             if (contentIdToLocationMapping != null)
             {
-                string queryString = context.Request.QueryString.HasValue ? context.Request.QueryString.Value : String.Empty;
-                string resolvedRequestUrl = ContentIdHelpers.ResolveContentId(queryString, contentIdToLocationMapping);
+                string displayUrl = context.Request.GetDisplayUrl();
```

Comment:
>GetDisplayUrl [](start = 52, length = 13)

Is GetDisplayUrl() the right method to use here?  Would GetEncodedUrl() be better?

Reply:
Updated to use GetEncodedUrl()

---
In reply to: [487340257](https://github.com/OData/WebApi/pull/2232#discussion_r487340257) [](ancestors = 487340257)